oidc-api/docs/AuthorisationHandler.md

```diff
@@ -0,0 +1,73 @@
+## Summary
+
+Main entrypoint into One Login.
+
+- Triggered by a call to /authorize from an RP
+
+  - RP must already be in the client registry
+
+- Possible journeys through lambda (can be found in [orchestration diagrams](../../docs/diagrams/orchestration))
+  - [Auth journey](../../docs/diagrams/orchestration/auth-only)
+  - [Doc App journey](../../docs/diagrams/orchestration/auth-only)
+
+## Functionality
+
+The lambda performs the following functions, although not all functions take place for every journey:
+
+- Handling sessions
+
+  - User session set or created - “parent session”
+  - New client session - effectively a child of user session
+  - Persistent session ID created if it doesn’t already exist
+  - All stored in redis
+
+- Validation
+
+  - Validates the client is in the client registry
+  - Some things specific to individual RP, so it’s about the value itself, e.g. permitted vtr or scopes from client registry
+  - Some things are general to all RPs for this lambda, mostly about the presence of particular params, e.g. must have a redirect_uri parameter
+
+- Generates redirect response
+
+  - Location header will determine where the browser should redirect to. In a success case the redirect url will be an oauth compliant authorize request.
+  - The auth params submitted here are converted to an encrypted jwt and will ‘follow’ an authentication flow potentially through many steps - in the first instance, they will be passed to auth frontend where they are mostly stored in the frontend session, but on their ‘way back’ they get stored in auth code and token store, and ultimately dictate what the userinfo response orch -> RP 
+  - Will contain Cookies in response
+
+## Examples
+
+### Input (for “normal” Auth journey) - will vary for other journeys:
+
+```json
+{
+"queryStringParams": {
+  "vtr": ["Cl"]
+  "scope": "openid email phone",
+  "claims": {"userinfo":{"https:\/\/vocab.account.gov.uk\/v1\/passport":{"essential":true},"https:\/\/vocab.account.gov.uk\/v1\/coreIdentityJWT":{"essential":true},"https:\/\/vocab.account.gov.uk\/v1\/address":{"essential":true}}},
```

Comment:
Can you have these claims with that vtr?

Reply:
I believe so since this data was taken from the auth stub